mkosi/installer/moss.py

```diff
@@ -0,0 +1,133 @@
+        return Path("moss")
+
+    @classmethod
+    def cache_subdirs(cls, cache: Path) -> list[Path]:
+        return [
+            p / "moss"
+            for p in cache.iterdir()
+            if p.is_dir() and "-" in p.name and "mkosi" not in p.name
+        ]
+
+    @classmethod
+    def scripts(cls, context: Context) -> dict[str, list[PathString]]:
+        return {
+            "moss": apivfs_cmd(context.root) + cls.cmd(context),
+            "mkosi-install": ["moss", "install"],
+            "mkosi-upgrade": ["moss", "sync"],
+            "mkosi-remove": ["moss", "remove"],
+        }
+
+    @classmethod
+    def mounts(cls, context: Context) -> list[PathString]:
+        return [
+            *super().mounts(context),
+            # Moss has all its cache in `,moss/cache`, mount that from the outside
+            "--bind",
+            _moss_cache_dir(context),
+            Path("/.moss/cache"),
+        ]
+
+    @classmethod
+    def setup(cls, context: Context, repositories: Iterable[Repository]) -> None:
+        # Create custom cache dir for moss
+        moss_cache_dir = _moss_cache_dir(context)
+        moss_cache_dir.mkdir(exist_ok=True)
+        (context.root / ".moss").mkdir(exist_ok=True)
+        (context.root / ".moss" / "cache").mkdir(exist_ok=True)
+
+        for repo in repositories:
+            cls.invoke(
+                context,
+                "repo",
+                [
+                    "add",
+                    "-c",
+                    repo.description,
+                    "-p",
+                    f"{repo.priority}",
+                    repo.id,
+                    repo.uri,
+                ],
+            )
+
+    @classmethod
+    def cmd(cls, context: Context) -> list[PathString]:
+        moss = cls.executable(context.config)
+        return [
+            moss,
+            "--directory",
+            context.root,
+            "--yes-all",
+        ]
+
+    @classmethod
+    def invoke(
+        cls,
+        context: Context,
+        operation: str,
+        options: Sequence[str] = (),
+        packages: Sequence[str] = (),
+        apivfs: bool = True,
+    ) -> None:
+        with finalize_source_mounts(
+            context.config,
+            ephemeral=os.getuid() == 0 and context.config.build_sources_ephemeral,
+        ) as sources:
+            run(
+                cls.cmd(context) + [operation, *options, *sort_packages(packages)],
+                sandbox=(
+                    context.sandbox(
+                        network=True,
+                        options=[
+                            "--bind",
+                            context.root,
+                            context.root,
+                            *cls.mounts(context),
+                            *sources,
+                            "--chdir",
+                            "/work/src",
+                        ],
+                    )
+                    + (apivfs_cmd(context.root) if apivfs else [])
```

Comment:
I expect that you just copy pasted this from another implementation, but it'd be great if we could figure out whether `moss` needs the APIVFS filesystems mounted at all when installing packages. I figure since it's trying to be modern it might be able to operate without these mounted.